### PR TITLE
Adds ability to encrypt data awaiting injestion.

### PR DESCRIPTION
**Pull request by absalon-james**
A random AES256 key is generated per environment per collection.
A uuid.uuid4() is generated per environment per collection.
Each collection archive will be stored as <data dir>/<uuid>.tar.gz
  while awaiting ingestion.

One time aes keys are generated with a vars plugin. The
keys are passed to tasks via environment variables. Sync has also
been altered to accept an AES256 key to decrypt a specific run.

Collection archives are no longer extracted for reading.
  The cloud_snitch.runs modules provides an interface for reading
  optionally encrypted run archives.

Sync will no longer operate on all collection runs in a directory.
  cloud-snitch-clean is no longer relevant and has been removed.
  cloud-snitch-sync concurrency is no longer relevant. This has
    moved to ansible forks. A collection run now generates a
    named set of data and sync will work only on the named set
    of data.

The syndication/sync role is now only for removing left over
  archives. These would be left over <uuid>.tar.gz encrypted
  archives left on disk in the event of an unexpected error.
  These archives are unusable without the one time keys.

### CODE REVIEW

File: syndication/action_plugins/sync.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import time
+from cloud_snitch.sync import sync_single
+
+from ansible.plugins.action import ActionBase
+
+
+class ActionModule(ActionBase):
+    """Runs sync on collected data locally.
+
+    Requires argumments uuid, and key
```
Comment (cfarquhar) [inline syndication/action_plugins/sync.py]:
```suggestion
    Requires arguments uuid and key
```

File: cloud_snitch/runs.py [hunk 2]
```diff
@@ -1,36 +1,157 @@
...
+
+    The archive must be .tar.gz
+
+    """
+    encoding = 'utf-8'
+
+    def __init__(self, filename, key=None, mode='r:gz'):
+        """Initialize the file object.
+
+        :param filename: Filename to read from or write to
```
[the remark this answers is not in the document]
Comment (absalon-james) [reply cloud_snitch/runs.py]: Good catch.

File: cloud_snitch/runs.py [hunk 3]
```diff
@@ -1,36 +1,157 @@
...
-    def _read_data(self):
-        """Reads run data.
+    def __init__(self, path, key=None):
+        """Inits the run
 
-        :returns: Run data loaded from file.
-        :rtype: dict
+        :param path: Path on disk that contains the run
+        :type path: str
+        :param key: Encyrption key to encrypt and decrypt run data
```
[the remark this answers is not in the document]
Comment (absalon-james) [reply cloud_snitch/runs.py]: Good catch

File: syndication/action_plugins/sync.py (hunk 1 above)
Comment (absalon-james) [reply syndication/action_plugins/sync.py]: Good catch

**Comment by absalon-james** [comment]
Fixed the mentioned comments and rebased to newer develop.

**Comment by chalupaul** [comment]
please merge this.

**Comment by chalupaul** [comment]
please merge this.

**Comment by cfarquhar** [comment]
merge this please

**Pull request merged**